game/src/level/elements/ILevel.java

```diff
@@ -211,4 +211,28 @@ default void changeTileElementType(Tile tile, LevelElement changeInto) {
             level.setEndTile(newTile);
         }
     }
+
+    @Override
+    default Tile getRandomTile(LevelElement elementType) {
+        return switch (elementType) {
+            case SKIP -> getSkipTiles().size() > 0
+                    ? getSkipTiles().get(RANDOM.nextInt(getSkipTiles().size()))
+                    : null;
+            case FLOOR -> getFloorTiles().size() > 0
+                    ? getFloorTiles().get(RANDOM.nextInt(getFloorTiles().size()))
+                    : null;
+            case WALL -> getWallTiles().size() > 0
+                    ? getWallTiles().get(RANDOM.nextInt(getWallTiles().size()))
+                    : null;
+            case HOLE -> getHoleTiles().size() > 0
+                    ? getHoleTiles().get(RANDOM.nextInt(getHoleTiles().size()))
+                    : null;
+            case EXIT -> getExitTiles().size() > 0
+                    ? getExitTiles().get(RANDOM.nextInt(getExitTiles().size()))
+                    : null;
+            case DOOR -> getDoorTiles().size() > 0
+                    ? getDoorTiles().get(RANDOM.nextInt(getDoorTiles().size()))
+                    : null;
+        };
```

Comment:
Ist okay, allerdings falls ein neues `LevelElement` hinzugefügt wird, muss man daran denken es hier hinzuzufügen...

Reply:
^^ ja um ehrlich zu sein stört mich das ganze auch. Das Problem ist dies ist das aktuelle umgehen mit den einzelnen Tiles, wieso dies so ist, weiß ich selbst nicht. Dafür würde ich lieber das einmal klären mit denen, die für die Planung zuständig sind. 

Um dann z.b. alle LevelElement Listen in eine Map zu packen und dann auch großflächig neue Tests zu bauen, um auch die Funktionalität zu gewährleisten, schätze ich kann länger dauern und somit würde dieser Bug länger existieren als vielleicht gut wäre.

Reply:
Wenn ich von außen drauf schaue, dann habe ich eine Unterscheidung basierend auf dem Wert von `LevelElement` und mache dann immer das Gleiche: Habe ich Elemente dieses Typs - dann hole ein zufälliges Element aus dieser Liste, sonst `null`.

Mal abgesehen von dem `null`, welches mich auf der nächsten Ebene wieder zwingt, einen `null`-Check zu machen (kann man das nicht gleich hier abhandeln?): der obige Ablauf sollte als _eine_ Funktion auf einer passenden Datenstruktur formuliert werden, die als Parameter den Wert `LevelElement` bekommt und damit in der Datenstruktur "nachschaut". Dann kann ich mir das `switch/case` sparen.

Reply:
Ja aktuell ist dieses gesamte bestimmte Tiles erstellen irgendwie allgemein unschön wir haben an sich eine Map<LevelElement, List<Tile>> welche alle Daten enthalten. 

Also programmiert wurde es als X ArrayLists welche dann per Hand gehandhabt werden.... Vielleicht nicht die beste Entscheidung finde ich. Dies wurde gemacht, damit die Level mit Türen verbunden werden konnten, aber dann nicht nochmal rüber iteriert, um es stimmig zu bekommen.

Warum ich es hier jetzt erstmal so lassen würde, wäre, damit wir nicht wieder Sachen in kleine Issues ziehen. Als Nächstes würde ich vorschlagen dann einmal das Interface ILevel sowie die Klasse TileLevel einmal refactorn, sodass dieses ´komische umgehen mit den LevelElementen wegfällt.

Reply:
Dann mach das wie vorgeschlagen: Lass den Code hier so und mache ein neues Ticket auf für das Refactoring mit konkreter Problembeschreibung, Link auf diese Diskussion sowie einer kurzen Beschreibung des möglichen Fixes.

Reply:
Issue #460 wurde dafür angelegt.